pkg/cache/v3/simple.go

```diff
@@ -547,20 +553,26 @@ func createResponse(snapshot ResourceSnapshot, watch ResponseWatch, ads bool) (*
 		}
 
 		for name := range knownResources {
-			_, exist := resources[name]
+			res, exist := resources[name]
 			if !exist {
 				deletedResources = append(deletedResources, name)
 				continue
 			}
 
 			if watch.subscription.IsWildcard() {
```

Comment:
I think we can do the same pattern here as for the delta wildcard requests


```suggestion
		// the only case where we can keep a resource if it's a not explicitly subscribed is when we are a wildcard subscription and the resource is also a wildcard resource.
		if watch.subscription.IsWildcard() && !res.OnDemandOnly() {
```

Reply:
Nice catch, updating